.github/actions/go-test-setup/action.yml

```diff
@@ -0,0 +1,30 @@
+name: setup-go-ipfs
+description: install and setup go-ipfs as needed for testing
+
+runs:
+  using: "composite"
+  steps:
+    - name: Install go-ipfs
+      shell: bash
+      run: (cd /tmp && go install github.com/ipfs/go-ipfs/cmd/ipfs@v0.9.1)
+    - name: Initialize go-ipfs
+      shell: bash
+      run: (ipfs init)
+    - name: Run go-ipfs
+      shell: bash
+      run: (ipfs daemon &)
+    - name: Wait for go-ipfs to be ready
+      shell: pwsh
```

Comment:
Does this work on Linux / Mac?

Reply:
Sorry about that. I've jumped at it to quickly and assumed it was already tested but it clearly doesn't look generic enough. Let me turn this back into a draft and retarget at the sync branch.